tests/NSubstitute.Acceptance.Specs/FieldReports/Issue533_TaskExceptionMightBeSwallowed.cs

```diff
@@ -0,0 +1,238 @@
+        {
+            var substitute = Substitute.ForPartsOf<ClassWithThrowingMethods>();
+
+            Assert.DoesNotThrow(() =>
+                substitute.GetValue().Returns(c => 42));
+        }
+
+        [Test]
+        public void ShouldSwallowCustomExceptionThrownFromTask_TaskCallbackAnyArg()
+        {
+            var substitute = Substitute.ForPartsOf<ClassWithThrowingMethods>();
+
+            Assert.DoesNotThrow(() =>
+                substitute.GetValue().ReturnsForAnyArgs(c => 42));
+        }
+
+        [Test]
+        public void ShouldSwallowCustomExceptionThrownFromTask_ValueTaskValue()
+        {
+            var substitute = Substitute.ForPartsOf<ClassWithThrowingMethods>();
+
+            Assert.DoesNotThrow(() =>
+                substitute.GetValueValueTask().Returns(42));
+        }
+
+        [Test]
+        public void ShouldSwallowCustomExceptionThrownFromTask_ValueTaskValueAnyArg()
+        {
+            var substitute = Substitute.ForPartsOf<ClassWithThrowingMethods>();
+
+            Assert.DoesNotThrow(() =>
+                substitute.GetValueValueTask().ReturnsForAnyArgs(42));
+        }
+
+        [Test]
+        public void ShouldSwallowCustomExceptionThrownFromTask_ValueTaskCallback()
+        {
+            var substitute = Substitute.ForPartsOf<ClassWithThrowingMethods>();
+
+            Assert.DoesNotThrow(() =>
+                substitute.GetValueValueTask().Returns(c => 42));
+        }
+
+        [Test]
+        public void ShouldSwallowCustomExceptionThrownFromTask_ValueTaskCallbackAnyArg()
+        {
+            var substitute = Substitute.ForPartsOf<ClassWithThrowingMethods>();
+
+            Assert.DoesNotThrow(() =>
+                substitute.GetValueValueTask().ReturnsForAnyArgs(c => 42));
+        }
+
+        private static async Task<string> InvokeEchoAsync(ISomething something, int value)
+        {
+            return await something.EchoAsync(value);
+        }
+
+        private static async ValueTask<string> InvokeEchoValueTaskAsync(ISomething something, int value)
+        {
+            return await something.EchoValueTaskAsync(value);
+        }
+
+        public interface IMainModule
+        {
+            IWorker Worker { get; }
+        }
+
+        public interface IWorker
+        {
+            Task DoSomethingAsync(Guid userId);
+        }
+
+        private static class MainModuleExtensions
+        {
+            public static async Task<Guid> Work(
+                /* this - extensions are not supported for nested types */ IMainModule mainModule,
+                Guid userId, string s)
+            {
+                await mainModule.Worker.DoSomethingAsync(userId);
+                return Guid.NewGuid();
+            }
+        }
+
+        public class ClassWithThrowingMethods
+        {
+#pragma warning disable 1998 // Async is required here to wrap exception in task.
+            public virtual async Task<int> GetValue() => throw new NotImplementedException();
```

Comment:
I think can use [`Task<int>.FromException`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.fromexception?view=netframework-4.8#System_Threading_Tasks_Task_FromException__1_System_Exception_) to avoid the pragma disable stuff here?

Reply:
This API is not available in .NET 4.5 unfortunately 😟